app/controllers/application_controller.rb

```diff
@@ -13,8 +13,21 @@ def logged_in?
 
   def require_user
     unless logged_in?
-      flash[:info] = "Access reserved for members only. Please sign in first."
+      flash[:error] = "Access reserved for members only. Please sign in first."
       redirect_to root_path
     end
   end
+
+  private
+
+  def set_reviews(object)
+    @reviews = object.reviews.includes(:user)
+    if @reviews.any?
+      total_rating = @reviews.map(&:rating).inject(&:+).to_f
+      @average_rating = ( total_rating / @reviews.size).round(1)
+    else
+      @average_rating = 0.0
+    end
+    @rating_chioces = Review::RATING_RANGE.map { |n| ["#{n} Stars", n]  }
```

Comment:
This functionality belongs to views and it fits to helpers, where you also have a method for it - `pluralize`